interoperability_report.py

```diff
@@ -359,461 +285,241 @@ def run_publisher_shape_main(
+        check_function <<in>>: function to check how the samples are received
+                by the Subscriber. By default it does not check anything.
+
+        The function runs several different processes: one for each Publisher
+        and one for each Subscriber shape_main application.
+        The number of processes depends on how many elements are in
+        the list of parameters.
+        Then it checks that the codes obtained are the expected ones.
     """
     log_message(f'run_test parameters: \
                     name_executable_pub: {name_executable_pub} \
                     name_executable_sub: {name_executable_sub} \
                     test_case: {test_case.name} \
-                    param_pub1: {param_pub1} \
-                    param_pub2: {param_pub2} \
-                    param_sub: {param_sub} \
-                    expected_code_pub1: {expected_code_pub1} \
-                    expected_code_pub2: {expected_code_pub2} \
-                    expected_code_sub: {expected_code_sub} \
+                    parameters: {parameters} \
+                    expected_codes: {expected_codes} \
                     verbosity: {verbosity} \
-                    timeout: {timeout}',
+                    timeout: {timeout} \
+                    check_function: {check_function.__name__}',
                     verbosity)
 
-    manager = Manager()
-    # used for storing the obtained ReturnCode
-    # (from Publisher 1, Publisher 2 and Subscriber)
-    code = manager.list(range(3))
-    data = Queue() # used for storing the samples
-
-    subscriber_finished = Event()
-    publisher_finished = Event()
+    # numbers of publishers/subscriber we will have. It depends on how
+    # many strings of parameters we have.
+    num_entities = len(parameters)
 
-    log_message('Creating temporary files', verbosity)
-    file_subscriber = tempfile.TemporaryFile(mode='w+t')
-    file_publisher1 = tempfile.TemporaryFile(mode='w+t')
-    file_publisher2 = tempfile.TemporaryFile(mode='w+t')
-
-    # Manager is a shared memory section where the three processes can access.
-    # return_code is a list of three elements where the different processes
-    # (publisher 1, publisher 2 and subscriber applications) copy their ReturnCode.
+    # Manager is a shared memory section where all processes can access.
+    # return_code is a list of elements where the different processes
+    # (publishers and subscribers shape_main applications) copy their ReturnCode.
     # These ReturnCodes are identified by the index within the list,
-    # every index identifies one application. Therefore, only one application
-    # must modifies one element of the list.
-    # Once both processes are finished, the list contains the ReturnCode
+    # every index identifies one shape_main application. Therefore, only one
+    # shape_main application must modifies one element of the list.
+    # Once all processes are finished, the list contains the ReturnCode
```

Comment:
```suggestion
    # Once all processes are finished, the list <name_of_variable> contains the ReturnCode
```

Please add the name of the parameter that will contain the ReturnCodes

Reply:
okey